### PR TITLE
Create a CMake target upon install

### PR DESCRIPTION
**Pull request by mheyman**
This updates the `CMakeLists.txt` file to include the ability to install. The install creates CMake target `flux::flux` for use in other CMake builds.

That is, you can build like:
```sh
cmake -S . -B build --install-prefix "$(realpath .)/install" -DFLUX_BUILD_EXAMPLES=Off -DFLUX_BUILD_TESTS=Off .
cmake --install build
```
and, in your consuming CMakeLists.txt, use like
```cmake
find_package(flux CONFIG REQUIRED PATHS "path/to/flux/install")
target_link_libraries(mytarget PRIVATE flux::flux)
```

The README has an update for this and there is an additional file used as part of the CMake target creation.

### CODE REVIEW

File: flux-config.cmake.in
Comment (tcbrindle) [inline flux-config.cmake.in:1]: Can this file be moved to the `cmake` directory?

File: CMakeLists.txt [hunk 1]
```diff
@@ -55,4 +61,57 @@ endif()
...
+set(PORT_NAME flux)
+set(CONFIG_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PORT_NAME}")
+
+# header-only doesn't need architeture differences so clear CMAKE_SIZEOF_VOIDP
+# temporarily when creating the version file.
+set(ORIGINAL_CMAKE_SIZEOF_VOIDP ${CMAKE_SIZEOF_VOIDP})
+set(CMAKE_SIZEOF_VOIDP "")
+write_basic_package_version_file(
+    "${PROJECT_BINARY_DIR}/${PORT_NAME}-version.cmake"
+    VERSION 0.0.1
```
Comment (tcbrindle) [inline CMakeLists.txt:75]: I haven't made any numbered releases of Flux, so I'm not sure where this version number comes from -- is it actually needed?

File: README.md
Comment (tcbrindle) [inline README.md:1]: Don't worry about changing the README file. Being able to install the project via standard CMake commands is useful, but not notable enough to warrant being on the front page.

File: flux-config.cmake.in
Comment (mheyman) [reply flux-config.cmake.in:1]: Will do.

File: CMakeLists.txt (hunk 1 above)
Comment (mheyman) [reply CMakeLists.txt:75]: You know, I don't think VERSION used to be required but I got a failure without it. I didn't try hard to not have one though. I'll look into it.

File: README.md
Comment (mheyman) [reply README.md:1]: 👍

File: CMakeLists.txt (hunk 1 above)
Comment (mheyman) [reply CMakeLists.txt:75]: The docs say

> [If no VERSION is given, the PROJECT_VERSION variable is used. If this hasn’t been set, it errors out.](https://cmake.org/cmake/help/v3.3/module/CMakePackageConfigHelpers.html#command:write_basic_package_version_file)

So, unfortunately, `VERSION` is actually needed.

I will set it to "-1" so even version 0 will be greater than it. Poking in the CMake source makes me believe that "-1" is valid where something like "(unset)" (with the "(" lexicographically less than "0") will fail do to CMake `math()` calls on it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -55,4 +61,63 @@ endif()
...
-endif()
\ No newline at end of file
+endif()
+
+set(PORT_NAME flux)
+set(CONFIG_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PORT_NAME}")
+
+set(FOO "-1")
+math (EXPR BAR "${FOO} + 1")
+message(STATUS "BAR=${BAR}")
```
Comment (tcbrindle) [inline CMakeLists.txt:71]: Are these lines needed?

File: README.md
Comment (tcbrindle) [inline README.md:1]: Leftover whitespace change

**Review by tcbrindle — COMMENTED** [review]
Thanks for the changes.

I've noticed a couple of CMake warnings in the CI runs:

```
CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at /usr/local/share/cmake-3.27/Modules/GNUInstallDirs.cmake:243 (message):
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  CMakeLists.txt:4 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The CXX compiler identification is GNU 13.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/linuxbrew/.linuxbrew/bin/g++-13 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at CMakeLists.txt:13 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "flux" has a relative path.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- BAR=0
```

The first one isn't related to this PR (but feel free to bump the required CMake version to 3.5 while you're working on things if you wish).

I guess for the second warning the `include(GnuInstallDirs)` line needs to come after `project(libflux CXX)`?

I'm not sure what needs to be done to fix the warning about relative vs absolute paths?

File: CMakeLists.txt (hunk 2 above)
Comment (mheyman) [reply CMakeLists.txt]: hahahaha - removed

**Review by tcbrindle — APPROVED** [review]

**Pull request merged**